### ISSUE TITLE
assist: replace generic with impl trait

### ISSUE THREAD

**Issue by pickx** [rust-lang/rust-analyzer#14626]
```rust
fn new<P: AsRef<Path>>(location: P) -> Self { }
```
Placing the cursor on `P: AsRef<Path>` or on the `P` of `location: P` should trigger an assist to change this to
```rust
fn new(location: impl AsRef<Path>) -> Self { }
```

We already have an assist `replace impl trait with generic` that does the opposite.
